cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -2572,6 +2572,249 @@
 (rule (lower (has_type (fits_in_64 ty) (uadd_overflow_trap a b tc)))
       (trap_if_overflow (add_with_flags_paired ty a b) tc))
 
+;;;; Helpers for `*_overflow` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; when true, sign extend, otherwise zero extend
+(decl put_in_reg_ext32 (Value bool) Reg)
```

Comment:
I changed it to use the `ArgumentExtension` enum which I thought is fitting enough. I also changed the `lower_extend_op` helper to use `ArgumentExtension` since that was using a `bool` before also.

Reply:
Much better, thank you!